lib/utils/utf8.c

```diff
@@ -79,3 +84,41 @@ char *utf8_lcpy(char *dst, const char *src, size_t n)
 
 	return dst;
 }
+
+bool utf8_is_valid(const unsigned char *str, size_t len)
+{
+	size_t i = 0, nbyte = 0;
+
+	/* It will also return false */
+	if (str == NULL) {
+		return false;
+	}
+
+	while (i < len) {
+		if (str[i] == '\0') {
+			break;
+		}
```

Comment:
Style: "`while (str[i] != '\0') {` is IMHO clearer and four lines shorter.

Reply:
Very good, I will change it. But I can't find where it is documented, and I'm not that good at writing documentation, if possible, can you help me write documentation for it?

Reply:
Doxygen comments go above the function declaration in the relevant header.  You can copy the format of existing ones (see include/zephyr/kernel.h for lots and lots and lots of examples).